src/openseadragon.js

```diff
@@ -749,7 +749,9 @@
 
 /* eslint-disable no-redeclare */
 function OpenSeadragon( options ){
-    return new OpenSeadragon.Viewer( options );
+    var viewer = new OpenSeadragon.Viewer( options );
+    OpenSeadragon._viewers.set(viewer.element, viewer);
```

Comment:
Not everyone creates viewers via this function... they can also just do `new OpenSeadragon.Viewer` themselves. Therefore setting the viewer into the map should happen inside the Viewer constructor.

Reply:
You're right, I've changed the registration in the `Viewer` constructor